File: native_library/mac/DesktopToastANE/SwiftOSXANE_bak/SwiftOSXANE.m
//
// Created by User on 04/12/2016.
// Copyright (c) 2016 Tua Rua Ltd. All rights reserved.
//
#import <Foundation/Foundation.h>


#include "SwiftOSXANE_oc.h"
#import "SwiftOSXANE-Swift.h"
#include <Adobe AIR/Adobe AIR.h>

SwiftOSXANE *swft;

#define FRE_FUNCTION(fn) FREObject (fn)(FREContext context, void* functionData, uint32_t argc, FREObject argv[])

// convert argv into a pointer array which can be passed to Swift
NSPointerArray * getFREargs(uint32_t argc, FREObject argv[]) {
    NSPointerArray * pa = [[NSPointerArray alloc] initWithOptions:NSPointerFunctionsOpaqueMemory];
    for (int i = 0; i < argc; ++i) {
        FREObject freObject;
        freObject = argv[i];
        [pa addPointer:freObject];
    }
    return pa;
}


FRE_FUNCTION (getAge) {
    return [swft getAgeWithArgv:getFREargs(argc, argv)];
}

FRE_FUNCTION(getPrice) {
    return [swft getPriceWithArgv:getFREargs(argc, argv)];
}

FRE_FUNCTION (getIsSwiftCool) {
    return [swft getIsSwiftCoolWithArgv:getFREargs(argc, argv)];
}


FRE_FUNCTION (getHelloWorld) {
    return [swft getHelloWorldWithArgv:getFREargs(argc, argv)];
}

void contextInitializer(void *extData, const uint8_t *ctxType, FREContext ctx, uint32_t *numFunctionsToSet, const FRENamedFunction **functionsToSet) {
    static FRENamedFunction extensionFunctions[] = {
        {(const uint8_t *) "getHelloWorld", NULL, &getHelloWorld},
        {(const uint8_t *) "getAge", NULL, &getAge},
        {(const uint8_t *) "getPrice", NULL, &getPrice},
        {(const uint8_t *) "getIsSwiftCool", NULL, &getIsSwiftCool}
    };

    *numFunctionsToSet = sizeof(extensionFunctions) / sizeof(FRENamedFunction);
    *functionsToSet = extensionFunctions;

    swft = [[SwiftOSXANE alloc] init];
    [swft setFREContextWithCtx:ctx];

}


void contextFinalizer(FREContext ctx) {
    return;
}

void TRSOAExtInizer(void **extData, FREContextInitializer *ctxInitializer, FREContextFinalizer *ctxFinalizer) {
    *ctxInitializer = &contextInitializer;
    *ctxFinalizer = &contextFinalizer;
}

void TRSOAExtFinizer(void *extData) {
    FREContext nullCTX;
    nullCTX = 0;
    contextFinalizer(nullCTX);
    return;
}

File: native_library/mac/DesktopToastANE/SwiftOSXANE_bak/SwiftOSXANE.swift
//
//  SwiftOSXANE.swift
//  SwiftOSXANE
//
//  Created by User on 04/12/2016.
//  Copyright © 2016 Tua Rua Ltd. All rights reserved.
//

import Cocoa
import Foundation

@objc class SwiftOSXANE: NSObject {

    private var dllContext: FREContext!
    private let aneHelper = ANEHelper()

    private func trace(value:String) {
        FREDispatchStatusEventAsync(self.dllContext, value, "TRACE")
    }

    func getIsSwiftCool(argv:NSPointerArray) -> FREObject {
        return aneHelper.getFreObject(bool: true)
    }

    func getPrice(argv:NSPointerArray) -> FREObject {
        return aneHelper.getFREObject(double: 59.99)
    }

    func getAgeWith(argv:NSPointerArray) -> FREObject {
        var age = 31
        let inFRE:FREObject! = argv.pointer(at: 0)
        let person:Dictionary<String, AnyObject> = aneHelper.getIdObjectFromFREObject(freObject: inFRE)
            as! Dictionary<String, AnyObject>
        if let val = person["age"] as? NSNumber { // AnyObject is read back as NSNumber
            age = Int(val) + 7
        }
        
        return aneHelper.getFreObject(int:age)
    }
    

    func getHelloWorld(argv:NSPointerArray) -> FREObject {
        let inFRE:FREObject! = argv.pointer(at: 0)
        let txt:String = aneHelper.getIdObjectFromFREObject(freObject: inFRE) as! String
        let fre:FREObject = aneHelper.createFREObject(className:"com.tuarua.Person")
        aneHelper.setFREObjectProperty(freObject: fre, name: "name", prop: aneHelper.getFreObject(string: "Kitty"))
        aneHelper.setFREObjectProperty(freObject: fre, name: "age", prop: aneHelper.getFreObject(int: 21))

        let person = aneHelper.getIdObjectFromFREObject(freObject: fre) as! Dictionary<String, AnyObject>
        let personName:String = person["name"] as! String

        return aneHelper.getFreObject(string: "\(txt) talking to \(personName)")

    }
    

    func setFREContext(ctx: FREContext) {
        dllContext = ctx
        aneHelper.setFREContext(ctx:ctx)
    }


}







